lib/virtual.js

```diff
@@ -75,14 +146,81 @@ VirtualModulesPlugin.prototype.writeModule = function(filePath, contents) {
 		birthtime: time
 	});
 
+	// When using the WatchIgnorePlugin (https://github.com/webpack/webpack/blob/52184b897f40c75560b3630e43ca642fcac7e2cf/lib/WatchIgnorePlugin.js),
+	// the original watchFileSystem is stored in `wfs`. The following "unwraps" the ignoring
+	// wrappers, giving us access to the "real" watchFileSystem.
+	let finalWatchFileSystem = this._watcher && this._watcher.watchFileSystem;
+
+	while (finalWatchFileSystem && finalWatchFileSystem.wfs) {
+		finalWatchFileSystem = finalWatchFileSystem.wfs;
+	}
 	this.compiler.inputFileSystem._writeVirtualFile(filePath, stats, contents);
 };
 
-function setData(storage, key, value) {
-	if (storage.data instanceof Map) {
-		storage.data.set(key, value);
+function getData(storage, key) {
+	// Webpack 5
+	if (storage._data instanceof Map) {
+		return storage._data.get(key);
+	} else if (storage._data) {
+		return storage.data[key];
+	} else if (storage.data instanceof Map) {
+		// Webpack v4
+		return storage.data.get(key);
+	} else {
+		return storage.data[key];
+	}
+}
+
+function setData(backendOrStorage, key, valueFactory) {
+	const value = valueFactory(backendOrStorage);
+
+	// Webpack v5
+	if (backendOrStorage._data instanceof Map) {
+		backendOrStorage._data.set(key, value);
+	} else if (backendOrStorage._data) {
+		backendOrStorage.data[key] = value;
+	} else if (backendOrStorage.data instanceof Map) {
+		// Webpack 4
+		backendOrStorage.data.set(key, value);
+		backendOrStorage.data.set(key, value);
+	} else {
+		backendOrStorage.data[key] = value;
+		backendOrStorage.data[key] = value;
+	}
+}
+
+function getStatStorage(fileSystem) {
+	if (fileSystem._statStorage) {
+		// Webpack v4
+		return fileSystem._statStorage;
+	} else if (fileSystem._statBackend) {
+		// webpack v5
+		return fileSystem._statBackend;
+	} else {
+		// Unknown version?
+		throw new Error('Couldn\'t find a stat storage');
+	}
+}
+
+function getFileStorage(fileSystem) {
+	if (fileSystem._readFileStorage) {
+		// Webpack v4
+		return fileSystem._readFileStorage;
+	} else if (fileSystem._readFileBackend) {
+		// Webpack v5
+		return fileSystem._readFileBackend;
+	} else {
+		throw new Error('Couldn\'t find a readFileStorage');
+	}
+}
+
+function getReadDirBackend(fileSystem) {
+	if (fileSystem._readdirBackend) {
+		return fileSystem._readdirBackend;
+	} else if (fileSystem._readdirStorage) {
+		return fileSystem._readdirStorage;
 	} else {
-		storage.data[key] = value;
+		throw new Error('Couldn\'t find a readDirStorage from Webpack Internals');
```

Comment:
I probably shouldn't have commented on this line. I missed that it has an apostrophe that needed to be escaped. We use double quotes when there's an apostrophe to avoid the need to escape. We've setup an [eslint config](https://github.com/sveltejs/eslint-config) on the main svelte repo to check for this stuff, but haven't set it up on this project yet, so sorry it's not quite as automated here yet